modules/alma/lib/AlmaClient/AlmaClient.class.php

```diff
@@ -410,6 +427,70 @@ private static function loan_sort($a, $b) {
   }
 
   /**
+   * Add user consent.
+   */
+  public function add_user_consent($borr_card, $pin_code, $type) {
+    // Initialise the query parameters with the current value from the
+    // reservation array.
+    $params = array(
+      'borrCard' => $borr_card,
+      'pinCode' => $pin_code,
+      'allowType' => $type,
+    );
+
+    try {
+      $doc = $this->request('patron/allow/add', $params);
+      $res_status = $doc->getElementsByTagName('status')->item(0)->getAttribute('value');
+      // Return error code when patron is blocked.
+      if ($res_status != 'ok') {
+        return ALMA_AUTH_BLOCKED;
+      }
+
+      // General catchall if status is not okay is to report failure.
+      if ($res_status == 'consentNotOk') {
+        return FALSE;
+      }
+    }
+    catch (AlmaClientConsentNotFound $e) {
+      return FALSE;
+    }
+
+    return $res_status;
```

Comment:
How about returning `TRUE` if the result status equals `ok`. It would simplify the signature of the method (it would always return a boolean) and it seems like that is what `alma.user_consent.inc` does anyway.